diffmpm/material.py

```diff
@@ -133,6 +134,123 @@ def compute_stress(self, dstrain):
         return dstrain * self.properties["E"]
 
 
+@register_pytree_node_class
+class Bingham(Material):
+    _props = (
+        "density",
+        "youngs_modulus",
+        "poisson_ratio",
+        "tau0",
+        "mu",
+        "critical_shear_rate",
+    )
+
+    # Passing ndim as an extra parameter for the material to work for both 1D and 2D case
+    def __init__(self, material_properties, ndim):
+        """
+        Create a Bingham material model.
+
+        Arguments
+        ---------
+        material_properties: dict
+            Dictionary with material properties. For Bingham
+        material, 'density','youngs_modulus','poisson_ratio','tau0','mu'
+        and 'critical_shear_rate' are required keys.
+
+        ndim: int
+            Dimension of the problem supports 1D and 2D
+        """
+        self.validate_props(material_properties)
+        self.ndim = ndim
+        density = material_properties["density"]
+        youngs_modulus = material_properties["youngs_modulus"]
+        poisson_ratio = material_properties["poisson_ratio"]
+        tau0 = material_properties["tau0"]
+        mu_ = material_properties["mu"]
+        critical_shear_rate = material_properties["critical_shear_rate"]
+        # Calculate the bulk modulus
+        bulk_modulus = youngs_modulus / (3.0 * (1.0 - 2.0 * poisson_ratio))
+        compressibility_multiplier_ = 1.0
+        # Special Material Properties
+        if "incompressible" in material_properties.keys():
+            incompressible = material_properties["incompressible"]
+            if incompressible:
+                compressibility_multiplier_ = 0.0
+        self.properties = {
+            **material_properties,
+            "bulk_modulus": bulk_modulus,
+            "compressibility_multiplier": compressibility_multiplier_,
+        }
+
+    def __repr__(self):
+        return f"Bingham(props={self.properties})"
+
+    # Initialise State Variables
+    def initialise_state_variables():
+        state_vars = {}
+        state_vars["pressure"] = 0.0
+        return state_vars
+
+    # State Variables
+    def state_variables():
+        return ["pressure"]
+
+    # Compute the pressure
+    def thermodynamic_pressure(self, volumetric_strain):
+        return -self.properties["bulk_modulus"] * volumetric_strain
+
+    # Compute the stress
+    def compute_stress(self, dstrain, particle, state_vars):
```

Comment:
Are you thinking of making `state_vars` a particle attribute that gets passed here? Because I did not see any call to `initialise_state_variables()`. Also,

```suggestion
    def compute_stress(self, dstrain, particles, state_vars):
```
as it is a container of particles and not a single particle.